src/Calculator.ts

```diff
@@ -433,3 +427,54 @@ export async function calculateGapsInCare(
 
   return { results: result, debugOutput };
 }
+
+export function calculateDataRequirements(
+  measureBundle: R4.IBundle
+): { results: R4.ILibrary; debugOutput?: DebugOutput } {
+  // Extract the library ELM, and the id of the root library, from the measure bundle
+  const { cqls, rootLibIdentifier, elmJSONs } = MeasureHelpers.extractLibrariesFromBundle(measureBundle);
+  const rootLib = elmJSONs.find(ej => ej.library.identifier == rootLibIdentifier);
+
+  // We need a root library to run dataRequirements properly. If we don't have one, error out.
+  if (!rootLib?.library) {
+    throw new Error("root library doesn't contain a library object");
+  }
+
+  // get the retrieves for every statement in the root library
+  const allRetrieves = rootLib.library.statements.def.flatMap(statement => {
+    if (statement.expression && statement.name != 'Patient') {
+      const retrieves = RetrievesHelper.findRetrieves(rootLib, elmJSONs, statement.expression);
+      return retrieves;
+    } else {
+      return [];
+    }
+  });
+  const uniqueRetrieves = uniqWith(allRetrieves, isEqual);
```

Comment:
This doesn't seem to properly filter and create unique data requirements. I tested with EXM124 `dbcg-connectathon/fhir401/bundles/measure/EXM124-9.0.000/EXM124-9.0.000-bundle.json` and found this entry twice: 
```
{
  "type": "Observation",
  "codeFilter": [
    {
      "valueSet": "http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.108.12.1017"
    }
  ]
}
```

Reply:
Ugh. I've been battling with uniq-ing these objects. Guess the fight is still going. Good catch

Reply:
I think a custom function is going to be needed here to check for unique retrieves. Another approach would be to run the unique operation on the list of DataRequirements after it's initially transformed from the list of retrieves.

Reply:
Yeah, I think you're correct here.